err_stash.py

```diff
@@ -625,6 +636,70 @@ def merge(
         yield "{x} dry-run {x}".format(x="-" * 30)
 
 
+def delete_branches(stash_api, github_api, branches_to_delete) -> Iterator[str]:
+    if len(branches_to_delete) > 0:
+        yield f"Deleting Branches `{branches_to_delete[0]['branch_name']}`:"
+        for branch in branches_to_delete:
+            if branch["comes_from_github"]:
+                github_api.delete_branch(
+                    organization=branch["project"],
+                    repo_name=branch["slug"],
+                    branch_name=branch["branch_name"],
+                )
+                yield f"Project: `{branch['project']}` - Repository: `{branch['slug']}` - From `GitHub` - ::"
+            else:
+                if branch['pull_request'] is not None:
+                    pr = stash_api.fetch_pull_request(branch["project"], branch["slug"], branch['pull_request'][0])
+                    pr.decline(branch['pull_request'][1])
+
+                stash_api.delete_branch(
+                    branch["project"],
+                    branch["slug"],
+                    branch["branch_name"],
+                )
+                yield f"Project: `{branch['project']}` - Repository: `{branch['slug']}` - From `Stash`"
+    else:
+        yield "No branches to delete."
+
+
+def obtain_branches_to_delete(
+    stash_api,
+    github_api,
+    stash_projects,
+    github_organizations,
+    branch_name,
+    branches_to_delete,
+) -> Iterator[str]:
+    try:
+        plans = create_plans(
+                stash_api,
+                github_api,
+                stash_projects,
+                github_organizations,
+                branch_name,
+                exactly_branch_name=True,
+                assure_has_prs=False
+            )
+    except CheckError as e:
+        yield "\n".join(e.lines)
+        raise StopIteration()
+
+    yield "Found branch `{}` in these repositories:".format(branch_name)
+    for plan in plans:
+        branches_to_delete.append(
+            {
+                "comes_from_github": plan.comes_from_github,
+                "project": plan.project,
+                "slug": plan.slug,
+                "branch_id": plan.branches[0]["latestCommit"],
+                "branch_name": plan.branches[0]["displayId"],
+                "pull_request": (plan.pull_requests[0]["id"], plan.pull_requests[0]["version"])if len(plan.pull_requests) > 0 else None
+            }
+        )
+        yield f"{'Stash' if not plan.comes_from_github else 'GitHub'}: {plan.slug.upper()} -> (commit id *{plan.branches[0]['latestCommit']}*) {'*HAS PR*' if len(plan.pull_requests) > 0 else ''}"
```

Comment:
Minor issue: I don't think we should use `upper` for the slug, to me it looks like we are screaming:
```
"GITHUB: FB-SIMBR-XXXX-STUFF -> (commit id **) *HAS PR*"
```

Reply:
Agreed